### ISSUE TITLE
VIEW does not work properly with Intersection types

### ISSUE THREAD

**Issue by RieksJ** [AmpersandTarski/Ampersand#291]
The `VIEW` statement (default view) does not always  work properly in the context of generalizations. The following script has a comment in the INTERFACE where this goes wrong. I still use Ampersand v3.2.0[HEAD:9f8f087].

```
CONTEXT Bug291 IN ENGLISH

RELATION seqName[Sequence*SeqName] [UNI]   REPRESENT SeqName TYPE ALPHANUMERIC
RELATION itemName[Item*ItemName] [UNI]         REPRESENT ItemName TYPE ALPHANUMERIC
RELATION itemInSeq[Item*Sequence] [UNI]    -- An Item can be part of one sequence
RELATION nextItem[Item*Item] [UNI,INJ,IRF] -- A SRC Item can have the TGT Item as its successor

CLASSIFY SeqTree IS Sequence /\ Item -- This enables trees of sequenced items.
seqTreeName :: SeqTree * SeqTreeName [UNI,TOT]
CLASSIFY SeqName ISA SeqTreeName
CLASSIFY ItemName ISA SeqTreeName

PURPOSE RULE "Congruence of SeqTree names"
{+Since Sequences and Items both have names, SeqTrees, which are both Sequences and Items, might have two (different) names. In order to ensure non-ambiguity of SeqTree naming, this rule ensures that the `itemName` of a SeqTree has the same value as the `seqName` of that SeqTree. The choice is based on the fact that `seqName`s always exist - it is a function, whereas `itemName`s do not (it is [UNI]).-}
ROLE ExecEngine MAINTAINS "Congruence of SeqTree names"
RULE "Congruence of SeqTree names": seqTreeName |- seqTreeName -- itemName;I[SeqTreeName] /\ seqName;I[SeqTreeName]
VIOLATION (TXT "{EX} InsPair;seqName;Sequence;", SRC I[Sequence], TXT ";SeqName;", TGT I[SeqName]
          ,TXT "{EX} InsPair;itemName;Item;", SRC I[Item], TXT ";ItemName;", TGT I[ItemName]
          )

POPULATION seqTreeName CONTAINS [ ("DocAtom", "Document") ]
  POPULATION seqTreeName CONTAINS [ ("Doc_1Atom", "Introduction") ]
  POPULATION seqTreeName CONTAINS [ ("Doc_2Atom", "Definitions") ]

  POPULATION itemInSeq CONTAINS [ ("Doc_1Atom", "DocAtom") ]
  POPULATION itemInSeq CONTAINS [ ("Doc_2Atom", "DocAtom") ]

  POPULATION nextItem CONTAINS [ ("Doc_1Atom", "Doc_2Atom") ]

VIEW SeqTrees: SeqTree(TXT "SeqTrees:", seqTreeName) --! This default view does not work !--
VIEW SeqTree: SeqTree { SeqTrees : seqTreeName } ENDVIEW
VIEW Item: Item { Item : seqTreeName } ENDVIEW

INTERFACE "Overview": V[SESSION*SeqTree];(I-(itemInSeq;itemInSeq~)) COLS
  [ "SeqTree" : I --! This is where the default view should work !--
  , "Details" : I COLS
     [ "Name" : I[SeqTree] <SeqTree>
     , "Items" : itemInSeq~ COLS
        [ "Prev" : nextItem~ <Item>
        , "Item" : I <Item>
        , "Next" : nextItem <Item>
  ]  ]  ]

ENDCONTEXT
```

**Comment by hanjoosten** [comment]
Please specify the complete version number. The version number 3.2 is incomplete and thus useless. Also, there is no need anymore not to use the current master.

**Comment by WolframKahl** [comment]
On Mon, Jan 25, 2016 at 11:15:10PM -0800, Rieks wrote:

```
> CONTEXT Bug291 IN ENGLISH
> 
> [...]
> RELATION itemInSeq[Item*Sequence] [UNI]    -- An Item can be part of one sequence
> RELATION nextItem[Item*Item] [UNI,INJ,IRF] -- A SRC Item can have the TGT Item as its successor
```

The ```--``` here presumable start comments.

```
> RULE "Congruence of SeqTree names": seqTreeName |- seqTreeName -- itemName;I[SeqTreeName] /\ seqName;I[SeqTreeName]
```

How about the ```--``` here?


Wolfram

**Comment by sjcjoosten** [comment]
also starts comments.
The only place where ``--`` does not start comments, is within a comment:

```
{-
this is a comment
-- this too -} but this is not ```

we use Haskell comments, so the ``{-`` and ``-}`` may be nested.

**Comment by stefjoosten** [comment]
Alas, nested comments do not work properly since we use `{+` and `-}` to delimit text in `PURPOSE` statements.

**Comment by stefjoosten** [comment]
By the way, I still have not diagnosed this bug, mainly because it has a low priority.

**Comment by Michiel-s** [comment]
I think I have a clue what's happening here.

The default population only inserts Atom ids in the `SeqTreeName` column of the `seqtreename` table. It looks like the Atoms `DocAtom`, `Doc_1Atom` and `Doc_2Atom` are not populated in the database. @hanjoosten could you have a look at this.

The view stuff is discussed and fixed in #323

**Comment by hanjoosten** [comment]
The problem with the original script, as supplied with this issue, is that there was more than one root concepts. (see issue #292). That issue has been solved, by throwing an error for that. Because of that, this script is not a valid script. 
Does this close this issue as well? @RieksJ : Please either close this ticket or supply with a script that is valid with the current version of Ampersand.
